lib/public/views/LhcFills/Overview/index.js

```diff
@@ -18,11 +18,26 @@ import { lhcFillsActiveColumns } from '../ActiveColumns/lhcFillsActiveColumns.js
 import { estimateDisplayableRowsCount } from '../../../utilities/estimateDisplayableRowsCount.js';
 import { paginationComponent } from '../../../components/Pagination/paginationComponent.js';
 import { frontLink } from '../../../components/common/navigation/frontLink.js';
+import { switchInput } from '../../../components/common/form/switchInput.js';
 
 const TABLEROW_HEIGHT = 46;
 // Estimate of the navbar and pagination elements height total; Needs to be updated in case of changes;
 const PAGE_USED_HEIGHT = 215;
 
+/**
+ * Display a toggle switch to display stable beams only
+ *
+ * @param {LhcFillsOverviewModel} lhcFillsOverviewModel the overview model
+ * @returns {Component} the toggle switch
+ */
+export const toggleStableBeamOnlyFilter = (lhcFillsOverviewModel) => {
+    const isStableBeamsOnly = lhcFillsOverviewModel.isStableBeamsOnly();
+    const onChange = isStableBeamsOnly
+        ? () => lhcFillsOverviewModel.setStableBeamsFilter(false)
+        : () => lhcFillsOverviewModel.setStableBeamsFilter(true);
+    return switchInput(isStableBeamsOnly, onChange, { labelAfter: 'STABLE BEAM ONLY' });
+};
+
```

Comment:
Here the onchange should not depend on external context, it should simply take as parameter the new state and set the filter accordingly

Reply:
Done

Reply:
Are you sure?